core/lib/utils/src/env.rs

```diff
@@ -0,0 +1,31 @@
+use std::{path::PathBuf, sync::Mutex};
+
+static WORKSPACE: Mutex<Option<PathBuf>> = Mutex::new(None);
```

Comment:
Nit: Could probably use a `OnceCell` here, using [`get_or_try_init`](https://docs.rs/once_cell/latest/once_cell/sync/struct.OnceCell.html#method.get_or_try_init) for synchronization (IIUC, these calls are linearized).